frb_codegen/src/config.rs

```diff
@@ -114,53 +194,81 @@ pub fn parse(raw: RawOpts) -> Opts {
-        skip_add_mod_to_lib: raw.skip_add_mod_to_lib,
-        llvm_path: raw.llvm_path.unwrap_or_else(|| {
-            vec![
-                "/opt/homebrew/opt/llvm".to_owned(), // Homebrew root
-                "/usr/local/opt/llvm".to_owned(),    // Homebrew x86-64 root
-                // Possible Linux LLVM roots
-                "/usr/lib/llvm-9".to_owned(),
-                "/usr/lib/llvm-10".to_owned(),
-                "/usr/lib/llvm-11".to_owned(),
-                "/usr/lib/llvm-12".to_owned(),
-                "/usr/lib/llvm-13".to_owned(),
-                "/usr/lib/llvm-14".to_owned(),
-                "/usr/lib/".to_owned(),
-                "/usr/lib64/".to_owned(),
-                "C:/Program Files/llvm".to_owned(), // Default on Windows
-                "C:/Program Files/LLVM".to_owned(),
-                "C:/msys64/mingw64".to_owned(), // https://packages.msys2.org/package/mingw-w64-x86_64-clang
-            ]
-        }),
-        llvm_compiler_opts: raw.llvm_compiler_opts.unwrap_or_else(|| "".to_string()),
-        manifest_path,
-        dart_root,
-        build_runner: !raw.no_build_runner,
-        exclude_sync_execution_mode_utility: raw
-            .exclude_sync_execution_mode_utility
-            .unwrap_or(false),
+    // exclude_sync_execution_mode_utility(s)
+    // (due to conflict of dart api of multiple blocks, keep it temporarily)
+    let exclude_sync_execution_mode_utilities = match raw.exclude_sync_execution_mode_utility {
+        Some(exclude_sync_execution_mode_utilities) => exclude_sync_execution_mode_utilities,
+        None => {
+            let mut exclude_sync_execution_mode_utilities = Vec::new();
+            for i in 0..rust_input_paths.len() {
+                if i == 0 {
+                    exclude_sync_execution_mode_utilities.push(false);
+                } else {
+                    exclude_sync_execution_mode_utilities.push(true);
+                }
+            }
+            exclude_sync_execution_mode_utilities
+        }
+    };
+
+    // build Opt for each rust api block
+    let mut opts = Vec::new();
+    for i in 0..rust_input_paths.len() {
+        opts.push(Opts {
+            rust_input_path: rust_input_paths[i].clone(),
+            dart_output_path: dart_output_paths[i].clone(),
+            dart_decl_output_path: raw
+                .dart_decl_output
+                .as_ref()
+                .map(|s| canon_path(s.as_str())),
+            c_output_path: c_output_paths.clone(), //same for all rust api blocks
+            rust_crate_dir: rust_crate_dirs[i].clone(),
+            rust_output_path: rust_output_paths[i].clone(),
+            class_name: class_names[i].clone(),
+            dart_format_line_length: raw.dart_format_line_length.unwrap_or(80),
+            skip_add_mod_to_lib: raw.skip_add_mod_to_lib, //same for all rust api blocks
+            llvm_path: raw.llvm_path.clone().unwrap_or_else(|| {
+                vec![
```

Comment:
Please extract this llvm default to a function

Reply:
Fixed in 9e38c2995f796d3e5c35b6bbaa631d62645325dc. New function is called `get_llvm_paths`.